### PR TITLE
[Impeller] fix validation error for playground texture upload.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Buffer to image is missing barriers. We don't really need to use this for playgrounds though, as set contents does the right thing more or less.

Fixes https://github.com/flutter/flutter/issues/142024

### CODE REVIEW

File: impeller/playground/playground.cc [hunk 1]
```diff
@@ -385,77 +385,43 @@ static std::shared_ptr<Texture> CreateTextureForDecompressedImage(
...
+    VALIDATION_LOG << "Could not allocate texture for fixture.";
+    return nullptr;
+  }
 
+  auto uploaded = texture->SetContents(decompressed_image.GetAllocation());
+  if (!uploaded) {
+    VALIDATION_LOG << "Could not upload texture to device memory for fixture.";
+    return nullptr;
+  }
+  if (enable_mipmapping) {
```
Comment (jonahwilliams) [inline impeller/playground/playground.cc:407]: This branch down here was missing the mipmapping support.

**Review by gaaclarke — CHANGES_REQUESTED** [review]
Okay, this makes sense.  We are just ripping out the SupportsBufferToTextureBlits path to make sure the image format is getting set correctly, right?

This needs a test (something that exercises it stays fixed on ci).  I think maybe the easiest thing to do would be for me to land the PR that turns on validation, but have a way to selectively turn it off for certain tests.  Then this PR can turn it back on for a handful of tests.  How's that sound?

**Comment by jonahwilliams** [comment]
Are there other validation errors? If not I think we land this and then follow up to turn on validations. Ultimately this _is_ just test code.

**Comment by gaaclarke** [comment]
> Are there other validation errors? If not I think we land this and then follow up to turn on validations. Ultimately this _is_ just test code.

Chinmay was running into validation errors too when getting the playgrounds working (not goldens) and he was running into some others.  My PR's CI runs got messed up, let me try to get them running and get back to you.  If this is the only validation error, just pull my patch into this PR.

**Comment by gaaclarke** [comment]
For reference: This is the draft PR that attempts to turn validation on for the golden image tests https://github.com/flutter/engine/pull/49955

**Comment by gaaclarke** [comment]
I ran into another failure later when validations are on:
```
[ RUN      ] Play/AiksTest.ClipRectElidesNoOpClips/Vulkan
[FATAL:flutter/impeller/renderer/backend/vulkan/resource_manager_vk.cc(26)] Check failed: waiter_.get_id() != std::this_thread::get_id(). The ResourceManager being destructed on its own spawned thread is a sign that ContextVK was not properly destroyed. A usual fix for this is to ensure that ContextVK is shutdown (i.e. context->Shutdown()) before the ResourceManager is destroyed (i.e. at the end of a test).
```

There is still something goofed up.  Running `ClipRectElidesNoOpClips` alone doesn't reproduce the crash.

Let me see if we can get validation on just for `CanRenderImageRect` so we can land this.

**Review by gaaclarke — APPROVED** [review]
I turned on the validations for one of the tests that would fail before this PR.  LGTM, thanks!

**Comment by jonahwilliams** [comment]
woohoo!

**Pull request merged**